elixir/series/lib/string_series.ex

```diff
@@ -13,9 +13,22 @@ defmodule StringSeries do
     cond do
       String.length(string) == n ->
         [string]
+
       String.length(string) < n ->
         []
+
+      String.length(string) > n ->
+        count = String.length(string) - n + 1
+        process_string(string, n, count)
     end
```

Comment:
`String.length` is being calculated multiple times here, this could be expressed as a `case` statement with guards as well.

```elixir
    case String.length(string) do
      ^n ->
        [string]

      length when length < n ->
        []

      length ->
        count = length - n + 1
        process_string(string, n, count)
    end
```

Reply:
I really like this suggestion! Thank you!